lib/chatkit/client.rb

```diff
@@ -769,11 +782,44 @@ def update_permissions_for_role(options)
       })
     end
 
+    def upload_attachment(token, room_id, part)
+      body = part[:file]
+      content_length = body.length
+
+      if content_length <= 0
+        raise Chatkit::MissingParameterError.new("File contents size must be greater than 0")
+      end
+
+      attachment_req = { content_type: part[:type],
+                         content_length: content_length
+                       }
+
+      attachment_response = api_request({
+        method: "POST",
+        path: "/rooms/#{CGI::escape room_id}/attachments",
+        body: attachment_req,
+        jwt: token
+      })
+
+      url = attachment_response[:body][:upload_url]
+      connection = Excon.new(url, :omit_default_port => true)
+      upload_response = connection.put
```

Comment:
Don't we need to specify the PUT body somewhere?

Reply:
:man_facepalming: 

thanks

Reply:
I guess this means there's a lack of a test. In Android, I've had to stub out some of tests though because I haven't implemented the relevant fetch method yet though, which isn't ideal.

Reply:
Yes, I added the test in the fetch PR: https://github.com/pusher/chatkit-server-ruby/pull/26 (and fixed it in this PR)